### ISSUE TITLE
Unable to send userauth-publickey request

### ISSUE THREAD

**Issue by radar** [libgit2/rugged#789]
Rugged version: 

```
# Gemfile
gem 'rugged'
# Gemfile.lock version
0.28.1
```

Compiled with threads, https + ssh listed as features.

## What I am trying to do:

I am trying to clone a public GitHub repo by using private credentials. 

```ruby
  creds = Rugged::Credentials::SshKey.new(
  username: 'git',
  publickey: File.expand_path("~/.ssh/id_rsa.pub"),
  privatekey: File.expand_path("~/.ssh/id_rsa"),
)

Rugged::Repository.clone_at(
  'git@github.com:radar/markdown_book_test.git',
  'markdown',
  credentials: creds
)
```

## What I expect to happen

The repository is cloned to the 'markdown' directory

## What actually happens

I get this error:

```
Rugged::SshError: Failed to authenticate SSH session: Unable to send userauth-publickey request
```

**Comment by radar** [comment]
**TL;DR: libssh2 1.8.1 == bad, libssh2 1.8.2 == good**

I noticed that on my other Mac here that its `libssh2` version was 1.8.0, and it was working fine. I upgraded it to libssh 1.8.1 and it broke in exactly the same way as Mac #1. On further investigation, seems like the libssh2 1.8.1 release has broken this exact feature and there's an issue about it here: https://github.com/libssh2/libssh2/issues/336.

I have just installed libssh2 1.8.2 from their master branch (via `brew install --HEAD libssh2`, which is targetting libssh2/libssh2@38bf7ce). There is a PR currently open on Homebrew/homebrew-core which will bump the libssh2 version to 1.8.2: https://github.com/Homebrew/homebrew-core/pull/38311 so that we don't have to run `brew install --HEAD`... that'll be merged in the next few days I bet.

I have just uninstalled & re-installed the `rugged` gem and now things work correctly:

```
>> require 'rugged'
=> true
>> Rugged.features
=> [:threads, :https, :ssh]
>> creds = Rugged::Credentials::SshKey.new(
?>   username: 'git',
?>   publickey: File.expand_path("~/.ssh/id_rsa.pub"),
?>   privatekey: File.expand_path("~/.ssh/id_rsa"),
?> )
=> #<Rugged::Credentials::SshKey:0x00007f7ee683be38 @username="git", @publickey="/Users/ryanbigg/.ssh/id_rsa.pub", @privatekey="/Users/ryanbigg/.ssh/id_rsa", @passphrase=nil>
>>
>> Rugged::Repository.clone_at(
?>   'git@github.com:radar/markdown_book_test.git',
?>   'markdown',
?>   ignore_cert_errors: true,
?>   credentials: creds
>> )
=> #<Rugged::Repository:70091513591380 {path: "/Users/ryanbigg/code/hanami/twist-v2/backend/markdown/.git/"}>
```
